.circleci/config.yml

```diff
@@ -79,56 +83,116 @@ aliases:
       - ~/watchman
     key: v1-watchman-v4.9.0
 
+  - &install-node-dependencies
+    |
+      npm config set spin=false
+      npm config set progress=false
+      npm install --no-package-lock
+
+  - &run-node-tests
+    | 
+      npm test -- --maxWorkers=2
+      npm run lint
+      npm run flow -- check
+
 defaults: &defaults
   working_directory: ~/react-native
 
 version: 2
 jobs:
+
+  # Runs unit tests on Node 8
   test-node-8:
     <<: *defaults
     docker:
       - image: circleci/node:8
     steps:
       - checkout
-      - run: npm install --no-package-lock
-      - run: |
-          npm test -- --maxWorkers=2
-          npm run lint
-          npm run flow -- check
-# eslint - doesn't run on non-PR builds
-      - run:
-          name: Analyze Code
-          command: |
-            if [ -n "$CIRCLE_PR_NUMBER" ]; then
-              npm install github@0.2.4
-              cat <(echo eslint; npm run lint --silent -- --format=json; echo flow; npm run flow --silent -- check --json) | GITHUB_TOKEN="af6ef0d15709bc91d""06a6217a5a826a226fb57b7" CI_USER=$CIRCLE_PROJECT_USERNAME CI_REPO=$CIRCLE_PROJECT_REPONAME PULL_REQUEST_NUMBER=$CIRCLE_PR_NUMBER node bots/code-analysis-bot.js
-            else 
-              echo "Skipping code analysis."
-            fi
+      - restore-cache: *restore-node-cache
+      - run: *install-node-dependencies
+      - save-cache: *save-node-cache
+      - run: *run-node-tests
 
+  # Runs unit tests on Node 6
   test-node-6:
     <<: *defaults
     docker:
       - image: circleci/node:6.11.0
     steps:
       - checkout
-      - run: npm install
-      - run: |
-          npm test -- --maxWorkers=2
-          npm run lint
-          npm run flow -- check
+      - restore-cache: *restore-node-cache
+      - run: *install-node-dependencies
```

Comment:
`--no-package-lock` is new to npm@5, which is not used in the Node 6 and Node 4 tests. It's why I initially didn't set up a npm install alias to use everywhere. I do think it's alright to use it here, because earlier npm versions will silently ignore the flag.